### ISSUE TITLE
Incorrect formatter fix when with single `msgAndArgs` argument

### ISSUE THREAD

**Issue by pgimalac** [Antonboom/testifylint#169]
Testify assertions can be called with a non-string value for `msgAndArgs` argument if there is a single such argument, as can be seen [in the code](https://github.com/stretchr/testify/blob/b074924938f86d417f1c9a845c7e8b0784d7f937/assert/assertions.go#L295) it will be rendered using `fmt.Sprintf("%+v")`.

This means that using `assert.Fail(t, "failed", true)` is functionally correct (this example is odd, but some cases can make sense, like using an error or some struct).

But when using `-formatter.require-f-funcs`, the formatter rule of `testifylint` will replace this with `assert.Failf(t, "failed", true)`, which doesn't compile as the first message argument of f-variant assertions must have type string.

Minimal reproducible example:
```go
package main

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestA(t *testing.T) {
	assert.Fail(t, "failed", true)
}
```

**Comment by ccoVeille** [comment]
Thanks for reporting this.

It's indeed a real bug that has consequences

We will take a look.

**Comment by ccoVeille** [comment]
I took a look. I found the piece of code involved.

I started adding the unit test.

I will have to find a way to detect such "uncommon" usage.

I'm very polite here, because the example you gave looks improper usage of testify.

But then, I'm unsure how to ignore/report it

Reporting it as invalid could be counter productive on large code base.

So for now, I think the better would be to simply ignore it

**Comment by ccoVeille** [comment]
Hey @pgimalac do you have any real life example available on GitHub that you could share about this ?

Thanks

**Comment by pgimalac** [comment]
I ran the linter on the https://github.com/DataDog/datadog-agent repository (over 1.1M LoC in Go, including 400k lines of test) and found dozens of occurrences (at least 30).

Most of them look like invalid uses of testify (eg. missing `failureMessage` arg of `Fail`, or using `Error` instead of `ErrorIs`), which I'll try to fix, but a few are still interesting:
- [this](https://github.com/DataDog/datadog-agent/blob/7c7f1f5e2f6e7a479e46f4fb2d2cf4a953e225c2/pkg/obfuscate/credit_cards_test.go#L254) uses a loop index as message
- [this](https://github.com/DataDog/datadog-agent/blob/7c7f1f5e2f6e7a479e46f4fb2d2cf4a953e225c2/pkg/util/log/log_test.go#L452) uses a whole testcase struct as message
- the best case I could find is [this one](https://github.com/DataDog/datadog-agent/blob/7c7f1f5e2f6e7a479e46f4fb2d2cf4a953e225c2/test/new-e2e/tests/windows/common/assert.go#L112): `msgAndArgs` is passed as a slice, so it works fine with `Fail` but doesn't compile with `Failf` which expects a string then a slice

Maybe a disabled-by-default rule to warn when the first `msgAndArgs` argument doesn't have type string could be interesting ?

**Comment by ccoVeille** [comment]
Thanks for pointing out the PR and context.

Please expect delays as most maintainers are in holidays (including me)

**Comment by pgimalac** [comment]
No worries I'm really not in a hurry, I can just disable the rule anyway !
